packages/server/lib/controllers/apiAuth.controller.ts

```diff
@@ -156,7 +157,8 @@ class ApiAuthController {
                     environment,
                     account,
                     auth_mode: 'API_KEY',
-                    operation: updatedConnection.operation
+                    operation: updatedConnection.operation,
+                    endUser: isConnectSession ? res.locals['endUser'] : undefined
```

Comment:
why not directly `res.locals['endUser']`? is it not `undefined` if not `isConnectSession`?

Reply:
true, I think I like the explicitness here and if (big if) I manage to fix the locals types `endUser` will be never in most case.